controllers/users/updateUserAvatar.js

```diff
@@ -0,0 +1,43 @@
+import path from "path";
+import fs from "fs/promises";
+import { v4 as uuidv4 } from "uuid";
+import { storeImage } from "../../config/multerConfing.js";
+
+import { isImageAndTransform } from "../../helpers/isImageAndTransform.js";
+import { updateUserAvatarURL } from "../../service/index.js";
+
+export async function updateUserAvatar(req, res, next) {
+  if (!req.file) {
+    return res.status(400).json({ message: "File isn't a photo" });
+  }
+
+  const { path: temporaryPath } = req.file;
+  const extension = path.extname(temporaryPath);
+  const fileName = `${uuidv4()}${extension}`;
+  const filePath = path.join(storeImage, fileName);
+
+  try {
+    await fs.rename(temporaryPath, filePath);
+  } catch (e) {
+    console.log(e);
+    await fs.unlink(temporaryPath);
+    return next(e);
+  }
+
+  const isValidAndTransform = await isImageAndTransform(filePath);
```

Comment:
W tym miejscu już powinnaś mieć pewność że obsługujesz obrazek. 

Dlatego Poniższa wiadomość powinna być dużo wcześniej. Robisz filtrację w poprzednim midelware, jeśli dojdziemy do updateStatusAvatar to na samym początku Trzeba sprawdzić czy wszystko jest okej z obrazkiem I jeżeli tak to pozwoli ci na dalszą jego obsługę, a jeżeli nie zwrócić informacje że coś jest nie tak z obrazkiem.

Reply:
Przerzuciłam tą część kodu na sam początek tej funkcji. Dziękuję za uwagę.